CHANGELOG.md

```diff
@@ -24,6 +24,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Requests with empty body and application/json Content-Type Header will now
   return 400 error instead of 500 error.
   [cyberark/conjur#1968](https://github.com/cyberark/conjur/issues/1968)
+- Changed how batch secret requests encode binary secrets into JSON.
+  Batch requests including a secret with binary data will now return properly
```

Comment:
This should probably be updated now, right?

I think this is now a "Changed" item instead of a "Fixed", and it should read more like
> When batch secret requests are sent with an `Accept: base64` header, the secret values in the response will all be Base64-encoded. Sending batch secret requests with this header enables users to retrieve Base64-encoded binary secret data from Conjur.

Which makes me wonder - do we want to also update this endpoint to send a more elegant response than:
```
[origin=10.128.2.70] [request_id=cd294a8e-8a69-47ea-a45a-3b192a24d077] [tid=32] Completed 500 Internal Server Error in 18ms
[origin=10.128.2.70] [request_id=cd294a8e-8a69-47ea-a45a-3b192a24d077] [tid=32]   
[origin=10.128.2.70] [request_id=cd294a8e-8a69-47ea-a45a-3b192a24d077] [tid=32] Encoding::UndefinedConversionError ("\x82" from ASCII-8BIT to UTF-8):
[origin=10.128.2.70] [request_id=cd294a8e-8a69-47ea-a45a-3b192a24d077] [tid=32]   
[origin=10.128.2.70] [request_id=cd294a8e-8a69-47ea-a45a-3b192a24d077] [tid=32] app/controllers/secrets_controller.rb:73:in `batch'
[origin=10.128.2.70] [request_id=cd294a8e-8a69-47ea-a45a-3b192a24d077] [tid=32] app/controllers/application_controller.rb:74:in `block in run_with_transaction'
[origin=10.128.2.70] [request_id=cd294a8e-8a69-47ea-a45a-3b192a24d077] [tid=32] app/controllers/application_controller.rb:73:in `run_with_transaction'
[origin=10.128.2.70] [request_id=cd294a8e-8a69-47ea-a45a-3b192a24d077] [tid=32] lib/rack/remove_request_parameters.rb:26:in `call'
[origin=10.128.2.70] [request_id=cd294a8e-8a69-47ea-a45a-3b192a24d077] [tid=32] lib/rack/default_content_type.rb:68:in `call'
```
when someone tries to retrieve binary secret data in a batch request without sending the `Accept: base64` header? (Sorry if this is already addressed in the PR - if so, it's not clear from the description). I'm not sure what the correct error response code would be.

Reply:
I just updated the PR description to clarify the error return code. It still does return a 500, however I added an error message which lets the user know what the actual issue is. I didn't want to make it one of the 400 responses because there is nothing wrong with the request itself, rather it is a server error. The return object is now something like this:

```json
{
  "error": {
    "code": "bad_secret_encoding",
    "message": "a string representation of the  ruby hash that failed to encode"
  }
}
```

Reply:
I think you could add a `Fixed` changelog entry that's something like
> Attempting to retrieve binary secret data in a batch secret request without using the new `Accept: Base64` header now returns an informative error message to explain that improper secret encoding is the cause of the 500 response.